### PR TITLE
878 jmx and jmx exporter configuration and authentification

### LINKED ISSUE #878: jmx and jmx-exporter configuration and authentification
## Configuration cleanup
In TDP, the jmx-exporter is configured as a javaagent (which is the recommended way) and communicates localy with the jmx
 ([see jmx documentation here](https://docs.oracle.com/javadb/10.10.1.2/adminguide/radminjmxenabledisable.html))

As such, only one option is needed for the jvm : `-Dcom.sun.management.jmxremote`. All other options are unneeded and should be deleted. These options are : 
- `-Dcom.sun.management.jmxremote.authenticate` 
- `-Dcom.sun.management.jmxremote.ssl`
- `-Dcom.sun.management.jmxremote.password.file`

Likewise, the options `ssl`, `username` and `password` specified in the jmx-exporter configuration file are unused to communicate with a  local jmx  

## Enabling ssl and authentification beetween prometheus and jmx-exporter.
- The jmx-exporter javaagent should be upgraded to a version that supports this feature. (0.20.00 for example)
- Agent and prometheus should be configured to use ssl and authentification [as explained in jmx-exporter documentation](https://github.com/prometheus/jmx_exporter/tree/release-0.20.0?tab=readme-ov-file#http-authentication-optional) 

## Secure each jmxexporter configuration file
* Today, the file is storing the password is world readable. Store configuration files in /etc/service with '600' mode

### PR DESCRIPTION
**Pull request by giem-git**
#### Which issue(s) this PR fixes

Fixes #878

#### Agreements

- [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: tdp_lib_dag/zookeeper.yml [hunk 1]
```diff
@@ -8,6 +8,10 @@
 - name: zookeeper_client_install
   depends_on: []
 
+- name: zookeeper_ssl-tls_install
```
Comment (rpignolet) [inline tdp_lib_dag/zookeeper.yml:11]: Must be added in `zookeeper_install` dependencies.

File: tdp_vars_defaults/hadoop/hadoop.yml [hunk 2]
```diff
@@ -31,28 +31,30 @@ hadoop_jhs_conf_dir: "{{ hadoop_root_conf_dir }}/conf.jhs"
...
-jmx_exporter_conf_dir: /etc/jmx-exporter
 
 # Hadoop JVM options
-jmx_common_opts: "-Dcom.sun.management.jmxremote=true -Dcom.sun.management.jmxremote.authenticate=true -Dcom.sun.management.jmxremote.ssl=true"
+jmx_common_opts: "-Dcom.sun.management.jmxremote=true"
 jaas_nn_opts: "-Djava.security.auth.login.config={{ hadoop_nn_conf_dir }}/krb5JAASnn.conf"
 jaas_rm_opts: "-Djava.security.auth.login.config={{ hadoop_rm_conf_dir }}/krb5JAASrm.conf"
 
+# jmx exporter configuration directory and file
+hdfs_jmx_exporter_conf_dir: "/etc/hdfs"
```
Comment (rpignolet) [inline tdp_vars_defaults/hadoop/hadoop.yml:41]: I think we do not need a directory for jmx exporter, we can use `hadoop_root_conf_dir`

File: tdp_lib_dag/zookeeper.yml (hunk 1 above)
Comment (giem-git) [reply tdp_lib_dag/zookeeper.yml:11]: Change commited

File: tdp_vars_defaults/hadoop/hadoop.yml (hunk 2 above)
Comment (giem-git) [reply tdp_vars_defaults/hadoop/hadoop.yml]: Changes commited : we now use `*_root_conf_dir` when available, otherwise `*_conf_dir`

**Pull request merged**